backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/button/internal/BpkButtonImpl.kt

```diff
@@ -150,12 +151,22 @@ internal fun ButtonIcon(
 }
 
 @Composable
-internal fun ButtonText(text: String, modifier: Modifier = Modifier) {
+internal fun ButtonText(
+    text: String,
+    type: BpkButtonType,
+    modifier: Modifier = Modifier,
+) {
+    val isLink = type == BpkButtonType.Link || type == BpkButtonType.LinkOnDark
     BpkText(
         text = text,
         modifier = modifier,
         maxLines = 1,
         overflow = TextOverflow.Ellipsis,
+        style = if (isLink) {
+            LocalTextStyle.current.copy(textDecoration = TextDecoration.Underline)
+        } else {
+            LocalTextStyle.current
+        },
```

Comment:
didn't use `BpkLink` here because it has its internal `LinkAnnotation.Clickable` which kept overriding the ripple effect of the button, so decided to just change the style instead of making lots of changes for displaying underline text decoration.